docs/howto/types.md

```diff
@@ -0,0 +1,65 @@
+(types)=
+# How to prepare your Go types 
+
+To use your Go types with SQLair you will need to do some preparation. SQLair
+supports inputs from structs maps and slices and can output into structs
+and maps. 
+
+In general, all types used with SQLair need to be named, otherwise there is no
+way to reference them in the query. The rest of the preparation of the type
+depends on what kind it is.
+
+## Structs
+
+To use a struct with SQLair you must tag all the fields that correspond to a
+database column with a `db` tag specifying the column name. The fields must be
+public (i.e. start with a capital letter), though the struct does not have to
+be. Any fields without a tag will be ignored.
+
+For example:
+```go
+type Person struct {
+    Name    string  `db:"name"`
+    ID      int     `db:"id_number"`
+    DoB     string  `db:"date_of_birth"`
+    Weight  int
+    height  int
+}
+```
+In this example the `Weight` and `height` fields will be ignored by SQLair.
+
+This struct and its corresponding columns can be referenced from a SQLair input
+or output expression.
+
+If a struct contains an embedded struct then SQLair will treat the fields of the
+embedded structs as if they were fields in the parent struct.
+
+## Maps
+
+Named maps with a base key type of string can be used with SQLair. The value
```

Comment:
Again, try to switch from Reference language (description) to How-to guide language (commands).